src/utils.ts

```diff
@@ -63,4 +65,112 @@ export namespace Utils {
+   *   2. Verifies that the number of operations in the transaction is equal to one and of type manageData.
+   *   3. Verifies if timeBounds are still valid.
+   *   4. Verifies if the transaction has been signed by the server and the client.
+   *   5. Verifies that the sequenceNumber is equal to zero.
+   *
+   *
+   * @see [SEP0010: Stellar Web Authentication](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md)
+   * @function
+   * @memberof Utils
+   * @param {string} challengeTx SEP0010 transaction challenge transaction in base64.
+   * @param {string} serverAccountID The server's stellar account.
+   * @example
+   * import { Utils, Network }  from 'stellar-sdk'
+   *
+   * Network.useTestNetwork();
+   *
+   * let challenge = Utils.verifyChallengeTx("base64tx", "server-account-id")
+   * @returns {boolean}
+   */
+  export function verifyChallengeTx(
+    challengeTx: string,
+    serverAccountId: string,
+  ): boolean {
+    const transaction = new Transaction(challengeTx);
+
+    const sequence = Number.parseInt(transaction.sequence, 10);
+
+    if (sequence !== 0) {
+      throw new InvalidSep10ChallengeError(
+        "The transaction sequence number should be zero",
+      );
+    }
+
+    if (transaction.source !== serverAccountId) {
+      throw new InvalidSep10ChallengeError(
+        "The transaction source account is not equal to the server's account",
+      );
+    }
+
+    if (transaction.operations.length !== 1) {
+      throw new InvalidSep10ChallengeError(
+        "The transaction should contain only one operation",
+      );
+    }
+
+    const [operation] = transaction.operations;
+
+    if (!operation.source) {
+      throw new InvalidSep10ChallengeError(
+        "The transaction's operation should contain a source account",
+      );
+    }
+
+    if (operation.type !== "manageData") {
+      throw new InvalidSep10ChallengeError(
+        "The transaction's operation should be manageData",
+      );
+    }
+
+    if (!signedBy(transaction, serverAccountId)) {
+      throw new InvalidSep10ChallengeError(
+        "The transaction is not signed by the server",
+      );
+    }
+
+    if (!signedBy(transaction, operation.source as string)) {
+      throw new InvalidSep10ChallengeError(
+        "The transaction is not signed by the client",
+      );
+    }
+
+    if (!validateTimebounds(transaction)) {
+      throw new InvalidSep10ChallengeError("The transaction has expired");
+    }
+
+    return true;
+  }
+
+  function validateTimebounds(transaction: Transaction): boolean {
+    if (!transaction.timeBounds) {
+      return false;
+    }
+
+    const now = Math.floor(Date.now() / 1000);
+    const { minTime, maxTime } = transaction.timeBounds;
+
+    return (
+      now >= Number.parseInt(minTime, 10) && now <= Number.parseInt(maxTime, 10)
+    );
+  }
+
+  function signedBy(transaction: Transaction, accountId: string): boolean {
```

Comment:
This function seems useful enough that it should not be local to `verifyChallengeTx` (and it should probably have a similar name structure too.)

Reply:
@morleyzhi done :) -- renamed it to `verifyTxSignedBy`, added documentation and made it public.